numpy/_core/src/multiarray/methods.c

```diff
@@ -1882,19 +1883,46 @@ array_reduce_ex_picklebuffer(PyArrayObject *self, int protocol)
     }
 
     /* Construct a PickleBuffer of the array */
-
-    if (!PyArray_IS_C_CONTIGUOUS((PyArrayObject*) self) &&
-         PyArray_IS_F_CONTIGUOUS((PyArrayObject*) self)) {
+    if (PyArray_IS_C_CONTIGUOUS((PyArrayObject *)self)) {
+        order = 'C';
+        picklebuf_args = Py_BuildValue("(O)", self);
+        rev_perm = Py_BuildValue("O", Py_None);
+    }
+    else if (PyArray_IS_F_CONTIGUOUS((PyArrayObject *)self)) {
         /* if the array if Fortran-contiguous and not C-contiguous,
          * the PickleBuffer instance will hold a view on the transpose
          * of the initial array, that is C-contiguous. */
         order = 'F';
-        transposed_array = PyArray_Transpose((PyArrayObject*)self, NULL);
+        transposed_array = PyArray_Transpose((PyArrayObject *)self, NULL);
         picklebuf_args = Py_BuildValue("(N)", transposed_array);
+        rev_perm = Py_BuildValue("O", Py_None);
     }
     else {
-        order = 'C';
-        picklebuf_args = Py_BuildValue("(O)", self);
+        order = 'K'; // won't be used
```

Comment:
```suggestion
        order = 'K';
```
You use it?  Could comment that it is ignored when unpickling (but right now it isn't although that isn't needed).

Reply:
thx, applied